### ISSUE TITLE
RegistryOptions is missing onigLib

### ISSUE THREAD

**Issue by WebFreak001** [microsoft/vscode-textmate#155]
I just upgraded from vscode-textmate 4.4.0 to 5.4.0 but now typescript complains I'm missing the `onigLib` property on my `vsctm.Registry` options.

Before I didn't have any IOnigLib instance anywhere so I don't know where I'm supposed to take this from.

The README example shows using some vscode-oniguruma library which I don't depend on however and I don't want to modify the default behavior of the library.

Is the property actually required?

**Comment by alexdima** [comment]
Yes, the breaking change from `4.x` to `5.x` is that the oniguruma library needs to be loaded by the user of `vscode-textmate`. So you need to provide to `vscode-textmate` an `IOnigLib` implementation. You can use `vscode-oniguruma` on your side to load up the WASM and provide that `IOnigLib` implementation to `vscode-textmate`. The change has been made because `vscode-textmate` is now an universal module (can be loaded both in nodejs and in the browser).
